src/permissions/PermissionController.test.ts

```diff
@@ -3923,16 +3913,14 @@ describe('PermissionController', () => {
 
       const controller = getDefaultPermissionController(options);
 
-      await expect(
-        async () =>
-          await controller.acceptPermissionsRequest({
-            metadata: { id, origin },
-            permissions: {
-              [PermissionNames.wallet_getSecretArray]: {},
-            },
-          }),
-      ).rejects.toThrow(new errors.PermissionsRequestNotFoundError(id));
+      await controller.acceptPermissionsRequest({
+        metadata: { id, origin },
+        permissions: {
+          [PermissionNames.wallet_getSecretArray]: {},
+        },
+      });
 
+      // Method is called only once
       expect(callActionSpy).toHaveBeenCalledTimes(1);
```

Comment:
Instead of testing that `call` is called only once, maybe it's worth testing that `ApprovalController:acceptRequest` is not called? I feel like that would match the test description better as otherwise it's a bit unclear.

Reply:
That check will be something like this

```
expect(callActionSpy).not.toHaveBeenNthCalledWith(
        1,
        'ApprovalController:acceptRequest',
        {
          id,
        },
      );
```

The above check looks better to me, as check for `not` can fail if either of parameter is different.

Reply:
Sorry I'm not sure I understand, why can't you say this?

```
expect(callActionSpy).not.toHaveBeenCalledWith(
        'ApprovalController:acceptRequest',
        {
          id,
        },
      );
```

Reply:
I can call above expect but since the parameter is an object it can fail an any of property is different.

But when I test equality with `'ApprovalController:hasRequest'` it is guaranteed that called action is not `'ApprovalController:acceptRequest'`